api/resources_portal/test/factories.py

```diff
@@ -86,3 +86,44 @@ def post(self, create, extracted, **kwargs):
         newOrg = Organization.objects.get(id=self.organization.id)
 
         assign_perm("add_members_and_manage_permissions", new, newOrg)
+
+
+class LeafGrantFactory(factory.django.DjangoModelFactory):
+    class Meta:
+        model = "resources_portal.Grant"
+
+    title = "Young Investigator's Grant"
+    funder_id = "1234567890"
+
+
+class GrantUserAssociationFactory(factory.django.DjangoModelFactory):
+    class Meta:
+        model = "resources_portal.GrantUserAssociation"
+
+    user = factory.SubFactory(UserFactory)
+    grant = factory.SubFactory(LeafGrantFactory)
+
+
+class GrantOrganizationAssociationFactory(factory.django.DjangoModelFactory):
+    class Meta:
+        model = "resources_portal.GrantOrganizationAssociation"
+
+    organization = factory.SubFactory(PersonalOrganizationFactory)
+    grant = factory.SubFactory(LeafGrantFactory)
+
+
+class GrantMaterialAssociationFactory(factory.django.DjangoModelFactory):
+    class Meta:
+        model = "resources_portal.GrantMaterialAssociation"
+
+    # Can I pass contact_user and organization along here?
```

Comment:
What do you mean here? Would you like contact_user and organization to be pulled out of your material factory? If you want to work with the SubFactory's data, you'll need to use the post generation hook, as seen here: https://factoryboy.readthedocs.io/en/v1.1.5/post_generation.html

Reply:
Sorry, I should have removed this already. I'm not exactly sure what I meant here, it was a note to myself from when I was first writing this.